### PR TITLE
Feature/integrate oidc

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

File: pkg/server/http/option.go [hunk 1]
```diff
@@ -6,17 +6,19 @@ import (
...
 type Options struct {
-	Logger  log.Logger
-	Context context.Context
-	Config  *config.Config
-	Metrics *metrics.Metrics
+	Logger        log.Logger
+	Context       context.Context
+	Config        *config.Config
+	Metrics       *metrics.Metrics
+	OpenIDConnect middleware.OIDCOptions
```
Comment (tboerger) [inline pkg/server/http/option.go]: Adding this to the options doesn't really make sense? :)

Comment (DeepDiver1975) [reply pkg/server/http/option.go]: Why? These are the config options of openid connect - these need to be set. 
If not here - where else could that be?

Comment (tboerger) [reply pkg/server/http/option.go]: but you won't add middleware options to the http server package. the middleware gets used within this package, but the command package or other packages using the http server should not even know about any middleware package.

Comment (DeepDiver1975) [reply pkg/server/http/option.go]: And where do I take the config from then? Just let me know and I implement this accordingly. THX

Comment (butonic) [reply pkg/server/http/option.go]: @tboerger So where do we configure the middleware? ocis-hello does it in https://github.com/owncloud/ocis-hello/blob/master/pkg/server/http/server.go#L46-L61

Comment (tboerger) [reply pkg/server/http/option.go]: @butonic what about https://github.com/owncloud/ocis-graph/blob/master/pkg/server/http/server.go#L30-L43?

**Review by butonic — APPROVED** [review]

File: reflex.conf [hunk 2]
```diff
@@ -1,2 +1,2 @@
 # backend
--r '^(cmd|pkg)/.*\.go$' -R '^node_modules/' -s -- sh -c 'make bin/ocis-graph-debug && bin/ocis-graph-debug --log-level debug server --debug-pprof --debug-zpages'
+-r '^(cmd|pkg)/.*\.go$' -R '^node_modules/' -s -- sh -c 'make bin/ocis-graph-debug && bin/ocis-graph-debug --log-level debug server --debug-pprof --debug-zpages --oidc-endpoint="https://deepdiver" --oidc-insecure=1'
```
Comment (tboerger) [inline reflex.conf:2]: https://deepdiver hardcoded? You know that you can export env variables before executing `make watch`?

Comment (DeepDiver1975) [reply reflex.conf:2]: :see_no_evil: 

we need to address the https handling in ocis (some proxy module or caddy or what ever) - then we can get rid of this.

Comment (tboerger) [reply reflex.conf:2]: You can still already avoid this hardcoded values by using env variables ;)

Comment (DeepDiver1975) [reply reflex.conf:2]: Sure. Will fix

**Pull request merged**